script/prerender/manage.ts

```diff
@@ -0,0 +1,397 @@
+
+  const allPagesQueuedPromise = asyncPool(
+    MAX_CONCURRENT_BOOKS, Object.entries(BOOKS), prepareAndQueueBook
+  );
+
+  // Wait for the workers stack to be ready
+  await waitUntilStackCreateComplete(
+    {client: cfnClient, maxWaitTime: WORKERS_DEPLOY_TIMEOUT_SECONDS, minDelay: 10, maxDelay: 10},
+    {StackName: WORKERS_STACK_NAME}
+  );
+
+  console.log('Retrieving queue URLs');
+
+  // Get the queue URLs
+  const describeStacksResult = await cfnClient.send(
+    new DescribeStacksCommand({StackName: WORKERS_STACK_NAME})
+  );
+
+  const stacks = assertDefined(
+    describeStacksResult.Stacks, '[CFN] [DescribeStacks] Unexpected response: missing Stacks key'
+  );
+  const stack = assertDefined(stacks[0], `${WORKERS_STACK_NAME} stack not found`);
+  const outputs = assertDefined(
+    stack.Outputs, '[CFN] [DescribeStacks] Unexpected response: missing stack Outputs key'
+  );
+
+  let workQueueUrl: string | undefined;
+  let deadLetterQueueUrl: string | undefined;
+
+  for (const output of outputs) {
+    const outputKey = assertDefined(
+      output.OutputKey,
+      '[CFN] [DescribeStacks] Unexpected response: missing output OutputKey key'
+    );
+    const outputValue = assertDefined(
+      output.OutputValue,
+      '[CFN] [DescribeStacks] Unexpected response: missing output OutputValue key'
+    );
+
+    switch (outputKey) {
+      case 'WorkQueueUrl':
+        workQueueUrl = outputValue;
+        resolveWorkQueuePromise(workQueueUrl);
+        break;
+      case 'DeadLetterQueueUrl':
+        deadLetterQueueUrl = outputValue;
+        break;
+    }
+  }
+
+  if (!workQueueUrl) {
+    throw new Error(`${WORKERS_STACK_NAME} stack did not have a WorkQueueUrl output`);
+  }
+
+  if (!deadLetterQueueUrl) {
+    throw new Error(`${WORKERS_STACK_NAME} stack did not have a DeadLetterQueueUrl output`);
+  }
+
+  console.log('Waiting for all jobs to be queued');
+
+  // Wait for all book pages to be queued
+  const books = await allPagesQueuedPromise;
+
+  numBooks = books.length;
+
+  console.log(`All ${numPages} page prerendering jobs and all ${numBooks} sitemap jobs queued`);
+
+  // console.log('Queuing sitemap index job');
+
+  await sqsClient.send(new SendMessageCommand({
+    MessageBody: JSON.stringify({ payload: { books }, type: 'sitemapIndex' }),
+    QueueUrl: workQueueUrl,
+  }));
+
+  console.log('1 sitemap index job queued');
+
+  const numJobs = numPages + numBooks + 1;
+
+  // Ensure the work queue is empty
+
+  console.log('Waiting 1 minute for the work queue attributes to stabilize');
+
+  // First wait 1 minute after sending the last message for the queue attributes to stabilize
+  // This is required according to SQS docs
+  await new Promise((resolve) => setTimeout(resolve, 60000));
+
+  console.log('Waiting for the work queue to be empty');
+
+  // Now check that all the NumberOfMessages attributes are 0
+  const getQueueAttributesCommand = new GetQueueAttributesCommand({
```

Comment:
this manage function is pretty big, i feel like you could extract this whole waiting for queue to be empty thing into a helper

Reply:
Split it into smaller pieces and as a bonus got rid of all non-const global variables in this file.